src/test/java/be/seeseemelk/mockbukkit/entity/EnderSignalMockTest.java

```diff
@@ -0,0 +1,111 @@
+package be.seeseemelk.mockbukkit.entity;
+
+import be.seeseemelk.mockbukkit.MockBukkitExtension;
+import be.seeseemelk.mockbukkit.MockBukkitInject;
+import be.seeseemelk.mockbukkit.ServerMock;
+import be.seeseemelk.mockbukkit.WorldMock;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.entity.EnderSignal;
+import org.bukkit.inventory.ItemStack;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(MockBukkitExtension.class)
+class EnderSignalMockTest
+{
+
+    @MockBukkitInject
+    private ServerMock server;
+    private EnderSignal enderSignal;
+    private WorldMock world;
+
+    @BeforeEach
+    void setUp()
+    {
+        world = server.addSimpleWorld("world");
+        enderSignal = new EnderSignalMock(server, UUID.randomUUID());
+        enderSignal.teleport(new Location(world, 0, 1, 0));
+    }
+
+    @Test
+    void testGetTargetLocationDefault()
+    {
+        assertThrows(IllegalStateException.class, () -> enderSignal.getTargetLocation());
+    }
+
+    @Test
+    void testSetTargetLocation()
+    {
+        Location location = new Location(world, 0, 0, 0);
+        enderSignal.setTargetLocation(location);
+        assertEquals(location, enderSignal.getTargetLocation());
+    }
+
+    @Test
+    void testSetTargetLocationAcrossWorlds()
+    {
+        Location location = new Location(server.addSimpleWorld("world2"), 0, 0, 0);
+        assertThrows(IllegalArgumentException.class, () -> enderSignal.setTargetLocation(location));
+    }
+
+    @Test
+    void testSetTargetLocationNull()
+    {
+        assertThrows(IllegalArgumentException.class, () -> enderSignal.setTargetLocation(null));
+    }
+
+    @Test
+    void testGetDropItemDefault()
+    {
+        assertTrue(enderSignal.getDropItem());
+    }
+
+    @Test
+    void testSetDropItem()
+    {
+        enderSignal.setDropItem(false);
+        assertFalse(enderSignal.getDropItem());
+    }
+
+    @Test
+    void testGetItemDefault()
+    {
+        assertEquals(Material.ENDER_EYE, enderSignal.getItem().getType());
+    }
+
+    @Test
+    void  testSetItem()
```

Comment:
Just a small detail:
```suggestion
    void testSetItem()
```